src/main/java/sg/ncl/MainController.java

```diff
@@ -2938,6 +2939,59 @@ public String adminEnergy(Model model,
         return "energy_usage";
     }
 
+    @RequestMapping("/admin/nodesStatus")
+    public String adminNodesStatus(Model model, HttpSession session) throws IOException {
+
+        if (!validateIfAdmin(session)) {
+            return NO_PERMISSION_PAGE;
+        }
+
+
+        // get number of active users and running experiments
+        Map<String, String> testbedStatsMap = getTestbedStats();
+        testbedStatsMap.put(USER_DASHBOARD_FREE_NODES, "0");
+        testbedStatsMap.put(USER_DASHBOARD_TOTAL_NODES, "0");
+
+        Map<String, List<Map<String, String>>> nodesStatus = getNodesStatus();
+        Map<String, Map<String, Long>> nodesStatusCount = new HashMap<>();
+
+        /*
+         loop through each of the machine type
+         tabulate the different nodes type
+         count the number of different nodes status, e.g. SYSTEMX = { FREE = 10, IN_USE = 11, ... }
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)